src/app/domain/entities/champion/champion-entity.ts

```diff
@@ -0,0 +1,31 @@
+import { DomainEntity } from '../base/domain-entity';
+
+export interface ChampionProps {
+  name: string;
+  key: string;
+  tags: string[];
+  stats: {
+    hp: number;
+    armor: number;
+    attackdamage: number;
+    attackspeed: number;
+  };
+}
+export class ChampionEntity extends DomainEntity {
+  public readonly name!: string;
+  public readonly tags!: string[];
+  public readonly stats!: {
+    hp: number;
+    armor: number;
+    attackdamage: number;
+    attackspeed: number;
+  };
```

Comment:
I think you should create an interface for this type.

Reply:
Sim foi criado, porem quando eu inicio essa variável o typescript força eu declara cada um dos atributos